### PR TITLE
Onboarding: Add weight tracking support to the Pet entity [golden]

### PR DESCRIPTION
**Pull request by Sadik-Bi**
Add weight tracking support #1

Task Description:
Objective
Add weight tracking support to the Pet entity in the Spring PetClinic REST API to enable veterinarians to monitor pet weight over time.

Requirements
REST API Updates: All existing Pet REST endpoints must include weight field in JSON: GET /api/pets/{petId} - return weight as decimal number in JSON response PUT /api/pets/{petId} - accept weight as decimal number in JSON request body GET /api/pets - return weight for all pets in response array POST /api/owners/{ownerId}/pets - accept weight when creating pets PUT /api/owners/{ownerId}/pets/{petId} - accept weight in updates JSON Format: Weight field should be represented as: Field name: weight
Type: number (decimal)
Optional: can be null for pets without recorded weight Example: "weight": 15.75 or "weight": null
FAIL_TO_PASS: org.springframework.samples.petclinic.rest.controller.PetRestControllerIntegrationTest, org.springframework.samples.petclinic.rest.controller.OwnerRestControllerTests

### CODE REVIEW

File: src/test/java/org/springframework/samples/petclinic/rest/controller/PetRestControllerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+        return mapper;
+    }
+
+    @Test
+    void testWeightFieldEndToEnd() throws Exception {
+        ObjectMapper mapper = createObjectMapper();
+
+
+        Owner owner = new Owner();
```
Comment (Gowrishbyndoor) [inline src/test/java/org/springframework/samples/petclinic/rest/controller/PetRestControllerIntegrationTest.java]: Integration tests should be like blackbox testing, so it should not have any dependencies to the implementation code (model or other), our integration tests should be call endpoint with proper data and validate the data.

File: src/test/java/org/springframework/samples/petclinic/rest/controller/PetRestControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+        mockMvc.perform(get("/api/pets")
+                .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$[?(@.id == " + petId + ")]").exists())
+            .andExpect(jsonPath("$[?(@.id == " + petId + ")].weight").value(hasItem(nullValue())));
+    }
+
+    @Test
+    void testWeightFieldThroughOwnerEndpoint() throws Exception {
```
Comment (Gowrishbyndoor) [inline src/test/java/org/springframework/samples/petclinic/rest/controller/PetRestControllerIntegrationTest.java]: Try to cover all the all the scenarios, status code 200, 400, 500 which ever applicable. 

example negative weight tests are missing.

**Review by Gowrishbyndoor — CHANGES_REQUESTED** [review]
Please fix the mentioned comment and once all PR check passed please update the tests with golden PR as well with latest tests.

**Comment by Sadik-Bi** [comment]
> Please fix the mentioned comment and once all PR check passed please update the tests with golden PR as well with latest tests.
@Gowrishbyndoor 
i have fixed the comment. but i didn't understand this feedback "please update the tests with golden PR as well with latest tests."

**Comment by DmitriyChernyakSetronica** [comment]
OwnerRestControllerTests is NOT an integration test—it's a unit test with mocks that doesn't provide full black-box testing.

The following cases still need to be added:

1. Creating a pet with a null weight
MISSING test: POST /api/owners/{ownerId}/pets with "weight": null

2. Persistence check WEAK check: testGetAllPetsIncludesWeightField_Success200() only checks for the presence of the field in the row, but does not check for specific values

3. Checking GET /api/owners/{ownerId}/pets/{petId}
MISSING test: No integration test for the GET /api/owners/{ownerId}/pets/{petId} endpoint with the weight field

4. Checking for updates via the owner endpoint with null
MISSING test: PUT /api/owners/{ownerId}/pets/{petId} with "weight": null

**Comment by Sadik-Bi** [comment]
@Gowrishbyndoor i have fixed the PR feedbacks but i don't see the task being active on Mindrift

**Comment by DmitriyChernyakSetronica** [comment]
@Sadik-Bi  Is this class TestDataUtils.java needed? Is there anything else extra?

**Comment by DmitriyChernyakSetronica** [comment]
Review these recommendations, correct the tests, or leave a comment if you find them unnecessary.
```
1. WEAK CHECK: testUpdateExistingPetWithNullWeight_Success204() (line 209)
Problem:

mockMvc.perform(put("/api/pets/1")
.content(updateJson)
.contentType(MediaType.APPLICATION_JSON)
.accept(MediaType.APPLICATION_JSON))
.andExpect(status().isNoContent());
// ❌ NO RESULT CHECK!

Missing verification:

It is NOT verified that weight is actually set to null.
A GET request must be added to verify persistence.
Suggested fix:

// Verify weight was set to null
mockMvc.perform(get("/api/pets/1")
.accept(MediaType.APPLICATION_JSON))
.andExpect(status().isOk())
.andExpect(jsonPath("$.weight").isEmpty());

2. MISSING BOUNDARY VALUES
Uncovered edge cases:

❌ Weight = 0.0 (minimum boundary value)
❌ Very large values ​​(e.g., 999.99 or 1000.0)
❌ High-precision decimal (e.g., 15.123456)
❌ Very small positive values ​​(e.g., 0.01)
These tests are CRITICAL for verifying:

Correctness of boundary value handling
Precision of decimal storage in the database
Validation of the minimum value
3. LOOSE NULL CHECK in testCreatePetWithNullWeightViaOwnerEndpoint_Success201() (line 609)
Current check:

.andExpect(jsonPath("$.weight").value((String) null))

Problem:

.value((String) null) may be an unstable check
Better to use .isEmpty() or .doesNotExist()
Suggested fix:

.andExpect(jsonPath("$.weight").isEmpty())

4. MISSING CHECK FOR GET /api/owners/{ownerId} WITH WEIGHT IN PETS
Scenario not covered:

GET /api/owners/{ownerId} returns a list of pets with a weight field
Need to check that nested pets contain weight
Suggested test:

@Test
void testGetOwnerIncludesPetsWithWeight_Success200() throws Exception {
// Check that GET /api/owners/{ownerId}
// returns pets with a weight field
}
```
